Makefile

```
@@ -78,54 +108,87 @@ shellcheck/install: $(BIN_DIR) guard/program/xz
 
 sh/%: FIND_SH := find . $(FIND_EXCLUDES) -name '*.sh' -type f -print0
+## Lints bash script files
 sh/lint: | guard/program/shellcheck
 	@ echo "[$@]: Linting shell scripts..."
 	$(FIND_SH) | $(XARGS) shellcheck {}
 	@ echo "[$@]: Shell scripts PASSED lint test!"
 
 json/%: FIND_JSON := find . $(FIND_EXCLUDES) -name '*.json' -type f
-json/lint: | guard/program/jq
+json/validate:
+	@ $(FIND_JSON) | $(XARGS) bash -c 'jq --indent 4 -S . "{}" > /dev/null 2>&1 || (echo "[{}]: Found invalid JSON file: "{}" "; exit 1)'
+	@ echo "[$@]: JSON files PASSED validation test!"
+
+## Lints json files
+json/lint: | guard/program/jq json/validate
 	@ echo "[$@]: Linting JSON files..."
-	$(FIND_JSON) | $(XARGS) bash -c 'cmp {} <(jq --indent 4 -S . {}) || (echo "[{}]: Failed JSON Lint Test"; exit 1)'
+	$(FIND_JSON) | $(XARGS) bash -c 'cmp {} <(jq --indent 4 -S . {})'
 	@ echo "[$@]: JSON files PASSED lint test!"
 
-json/format: | guard/program/jq
+## Formats json files
+json/format: | guard/program/jq json/validate
 	@ echo "[$@]: Formatting JSON files..."
 	$(FIND_JSON) | $(XARGS) bash -c 'echo "$$(jq --indent 4 -S . "{}")" > "{}"'
 	@ echo "[$@]: Successfully formatted JSON files!"
 
 tfdocs-awk/install: $(BIN_DIR)
-tfdocs-awk/install: ARCHIVE := https://github.com/plus3it/tfdocs-awk/archive/0.0.0.tar.gz
+tfdocs-awk/install: ARCHIVE := https://github.com/plus3it/tfdocs-awk/archive/0.0.2.tar.gz
 tfdocs-awk/install:
 	$(CURL) $(ARCHIVE) | tar -C $(BIN_DIR) --strip-components=1 --wildcards '*.sh' --wildcards '*.awk' -xzvf -
 
+## Generates terraform documentation
 docs/generate: | tfdocs-awk/install guard/program/terraform-docs
 	@ echo "[$@]: Creating documentation files.."
 	@ bash -eu -o pipefail autodocs.sh -g
 	@ echo "[$@]: Documentation generated!"
 
+## Lints terraform documentation
 docs/lint: | tfdocs-awk/install guard/program/terraform-docs
 	@ echo "[$@] Linting documentation files.."
 	@ bash -eu -o pipefail autodocs.sh -l
 	@ echo "[$@] documentation linting complete!"
 
+TERRAFORM_TEST_DIR ?= tests
 terratest/install: | guard/program/go
-	cd tests && go mod init tardigrade-ci/tests
-	cd tests && go build ./...
-	cd tests && go mod tidy
+	@ echo "[$@] Installing terratest"
+	cd $(TERRAFORM_TEST_DIR) && go mod init tardigarde-ci/tests
+	cd $(TERRAFORM_TEST_DIR) && go build ./...
+	cd $(TERRAFORM_TEST_DIR) && go mod tidy
+	@ echo "[$@]: Completed successfully!"
 
 terratest/test: | guard/program/go
-	cd tests && go test -count=1 -timeout 20m
+	@ echo "[$@] Starting Terraform tests"
+	cd $(TERRAFORM_TEST_DIR) && go test -count=1 -timeout 20m
+	@ echo "[$@]: Completed successfully!"
 
+## Runs terraform tests in the tests directory
 test: terratest/test
+
+BATS_RELEASE ?= 1.1.0
+bats/install: $(BIN_DIR)
+bats/install: ARCHIVE := https://github.com/bats-core/bats-core/archive/v$(BATS_RELEASE).tar.gz
+bats/install:
+	$(CURL) $(ARCHIVE) | tar -C $(BIN_DIR) -xzvf - > /dev/null 2>&1 || (echo "[$@]: Download failed"; exit 1)
```

Nice. `bats` doesn't provide binaries in gh-releases. Can still use the underlying concept to retrieve the tar.gz bundle url.

```suggestion
bats/install: BATS_VERSION ?= latest
bats/install:
	$CURL $(shell $(CURL) https://api.github.com/repos/bats-core/bats-core/releases/$(BATS_VERSION) | jq -r '.tarball_url') | tar -C $(TMP) -xzvf -
	$(TMP)/bats-core-*/install.sh ~
	bats --version
	rm -rf $(TMP)/bats-core-*
```